src/exoticatechnologies/modifications/exotics/impl/HullmodExotic.kt

```diff
@@ -29,24 +32,54 @@ open class HullmodExotic(
             val moduleSlotList = member.variant.moduleSlots
+            logger.info("onInstall()\tmoduleSlots: ${moduleSlotList}")
+            if (moduleSlotList == null || moduleSlotList.isEmpty()) return
+
+            // Print out the module slot list, and generate a small map of
+            // <this exotic>, <parent FleetMember> + <list of expected variants> + <list of variants we installed on>
+            val mutableVariantList = moduleSlotList
+                    .map { slot -> member.variant.getModuleVariant(slot) }
+                    .toMutableList()
+            // Obviously, add the 'member' variant to the list as well
+            mutableVariantList.add(member.variant)
+            // Now forget about the mutable version and use a immutable version
+            val variantList = mutableVariantList.toList()
+
+            // Iterate through each slot, getting their variant and trying to install there
             for (slot in moduleSlotList) {
                 val moduleVariant = member.variant.getModuleVariant(slot)
                 logger.info("onInstall()\tslot: ${slot}\tmoduleVariant: ${moduleVariant}")
                 if (moduleVariant == null) continue
                 val mods = get(member, moduleVariant)
                 logger.info("onInstall()\tmods: ${mods}")
                 mods?.let { nonNullMods ->
-                    logger.info("onInstall()\t--> installWorkaround()\tmember: ${member}, moduleVariant: ${moduleVariant}, mods: ${mods}, exotic: ${this}")
-                    installWorkaround(member, moduleVariant, nonNullMods, this)
-                    logger.info("onInstall()\t--> installHullmodOnVariant()\tmoduleVariant: ${moduleVariant}")
-                    installHullmodOnVariant(moduleVariant)
+                    if (HullmodExoticHandler.shouldInstallHullmodExoticToVariant(
+                            hullmodExotic = this,
+                            parentFleetMember = member,
+                            variant = moduleVariant,
+                            variantList = Optional.of(variantList)
+                    )) {
+                        logger.info("onInstall()\t--> installWorkaround()\tmember: ${member}, moduleVariant: ${moduleVariant}, mods: ${mods}, exotic: ${this}")
+                        installWorkaround(member, moduleVariant, nonNullMods, this)
+                        logger.info("onInstall()\t--> installHullmodOnVariant()\tmoduleVariant: ${moduleVariant}")
+                        installHullmodOnVariant(moduleVariant)
+                        // And mark it as installed in the HullmodExoticHandler
+                        HullmodExoticHandler.installHullmodExoticToVariant(
+                                hullmodExotic = this,
+                                parentFleetMember = member,     //oh lets hope this one works out ...
+                                variant = moduleVariant
+                        )
+                    }
```

Comment:
somehow, i believe, all of this can be extracted into `HullmodExoticHandler`

Reply:
probably can. question is, should it be?

Reply:
aha, so looking at the comment below, this is probably what i wanted to extract as `HullmodExoticHandler::installOnVariant(...)` ?

Reply:
we're merely using the HullmodExoticHandler's "flows" in this class. none of this is no longer relevant. resolving.